testing/go/alter_table_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package _go
+
+import (
+	"testing"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+func TestAlterTableAddPrimaryKey(t *testing.T) {
+	RunScripts(t, []ScriptTest{
+		{
+			Name: "Add Primary Key",
+			SetUpScript: []string{
+				"CREATE TABLE test1 (a INT, b INT);",
+				"CREATE TABLE test2 (a INT, b INT, c INT);",
+				"CREATE TABLE pkTable1 (a INT PRIMARY KEY);",
+			},
+			Assertions: []ScriptTestAssertion{
+				{
+					Query:    "ALTER TABLE test1 ADD PRIMARY KEY (a);",
+					Expected: []sql.Row{},
+				},
+				{
+					// Test the pk by inserting a duplicate value
+					Query:       "INSERT into test1 values (1, 2), (1, 3);",
+					ExpectedErr: "ERROR: duplicate primary key given: [1] (errno 1062) (sqlstate HY000) (SQLSTATE XX000)",
```

Comment:
We shouldn't be too specific with the error messages, since `(errno 1062) (sqlstate HY000) (SQLSTATE XX000)` is all remnant MySQL stuff that we'll eventually need to remove, and we don't want to have to change all of our tests at that time.

For reference, Postgres returns:
```
duplicate key value violates unique constraint "test1_pkey"
```
We're going to have to change this error anyway, but we could at least shorten it to `duplicate primary key given`. We use `strings.Contains` for the `ExpectedErr` for that reason.

Reply:
Oh nice, for some reason I thought this had to be an exact match on the full error message. Thanks for pointing out it's using `strings.Contains`.

Reply:
Fixed in 2a516b566c64cd1c9a8764b210ef2d8003fa344c